src/extensions.rs

```diff
@@ -270,6 +271,78 @@ impl RatchetTreeExtension {
     }
 }
 
+#[repr(u32)]
+#[cfg_attr(rustfmt, rustfmt_skip)]
+pub enum DeviceCapabilityType {
+    AddCap          = 0b0001u32,
+    RemoveCap       = 0b0010u32,
+    NonRemovableCap = 0b0100u32,
+    SelfUpdateCap   = 0b1000u32,
+}
+
+#[derive(Debug, Clone, Copy, PartialEq)]
+pub struct DeviceCapabilities(pub u32);
```

Comment:
I wonder why we need `DeviceCapabilities` as a struct. Isn't the `enum` enough?

Reply:
This is all existing code from the original MI implementation. I think the idea was that the the `u32` is used to encode some combination of `DevicecapabilityTypes`. So if we do it like that, I think we need both the `DeviceCapabilities` struct _and_ the enum. Is that what you meant? I'm not sure if this is the best way to implement this kind of pattern.

Reply:
I guess `DeviceCapabilities` should probably use a Set of `DeviceCapabilityType`.

Reply:
The disadvantage being that (if I understand correctly), enums can't represent HashSet instances :-(

Reply:
Going forward we should also see to what extent we want to have custom extensions in maelstrom. For testing purposes there is no problem, but ideally maelstrom should be as generic as possible.

Reply:
Maybe it's better if I leave all this stuff on my own fork/branch for now. I can still do the PR at a later date.

Reply:
> Going forward we should also see to what extent we want to have custom extensions in maelstrom. For testing purposes there is no problem, but ideally maelstrom should be as generic as possible.

We should add non-standard things behind feature flags then it's ok to have them in.